### PR TITLE
[add] Navbar Component for system wide routing

### PR DESCRIPTION
**Pull request by edbella**
I've added the Navbar Component and created a global routing in the App.js file.
I've also removed some previously installed packages, jest, enzyme and so on as create-react-app comes bundled with jest.

### CODE REVIEW

**Comment by edbella** [comment]
@Segun-Ade @kechyy @ChineduOguejiofor @justiceotuya 
The navbar component is setup but the test and build is failing because my next component push which is the Homepage component already exists in the App.js file and since I'm trying to limit the number of lines of code you'd have to review, I add to break it up.

File: src/components/navbar/components/Navbar.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import { Menu, Row, Col } from "antd";
+import Logo from "../images/placeholder-logo.svg";
+
+/**
+ * This is th Navbar component that spans the entire App.
+ * It is a top level element and as such it is resident in the App.js file which is the entry system into the root index.js
+ */
+
+class Navbar extends Component {
+  state = {
```
Comment (justiceotuya) [inline src/components/navbar/components/Navbar.jsx]: For this project we want you to use hooks. please convert to hooks

File: src/components/navbar/components/Navbar.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        <Row type="flex" justify="center">
+          <Col span={24}>
+            <div className="navbar" data-aos="slide-down">
+              <Menu
+                className="navbar-group"
+                onClick={this.handleClick}
+                selectedKeys={[this.state.current]}
+                mode="horizontal"
+              >
+                <Menu.Item className="nav-logo">
```
Comment (justiceotuya) [inline src/components/navbar/components/Navbar.jsx]: please write dry codes. you are repeating yourself with menu.item. 
if you notice all the Menu.Item are somewhat similar. so you need to make it dry. i will do this

**constant.js**
```
const menuItems = [
{
    menuClassName : 'nav-logo';
    linkTo: '/';
    imgSrc :  Logo //you can import the logo here;
},
{
    key: 'home';
    linkTo: '/';
}
]
```

**Comment by edbella** [comment]
@justiceotuya @enyeInc @Segun-Ade @kechyy @ChineduOguejiofor Please review

File: src/components/navbar/components/Navbar.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+const Navbar = () => {
+  const [currentNav, setCurrentNav] = useState("home");
+
+  /**
+   * @constant
+   * @listens {event} to dtermine which of the navbar element is active using the "key" parameter on the menu item
+   */
+  const handleClick = e => {
+    console.log("click ", e);
```
Comment (justiceotuya) [inline src/components/navbar/components/Navbar.jsx]: remove this

File: src/components/navbar/components/Navbar.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      <Row type="flex" justify="center">
+        <Col span={24}>
+          <div className="navbar" data-aos="slide-down">
+            <Menu
+              className="navbar-group"
+              onClick={handleClick}
+              selectedKeys={[currentNav]}
+              mode="horizontal"
+            >
+              {menuItems.map(menu => {
```
Comment (justiceotuya) [inline src/components/navbar/components/Navbar.jsx]: distructure

File: src/components/navbar/components/Navbar.jsx (hunk 3 above)
Comment (edbella) [reply src/components/navbar/components/Navbar.jsx]: Ok.

File: src/App.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
 import React, { Component } from "react";
-import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
+import { BrowserRouter as Router } from "react-router-dom";
 import AOS from "aos";
 import "aos/dist/aos.css";
+import "./App.scss";
```
Comment (Way2nnadi) [inline src/App.js]: import these css file in the `index.js` file

File: src/components/navbar/components/Navbar.test.jsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { BrowserRouter as Router } from "react-router-dom";
+import Navbar from "./index";
+
+it("renders without crashing", () => {
+  const div = document.createElement("div");
+  ReactDOM.render(
+    <Router>
+      {" "}
```
Comment (Way2nnadi) [inline src/components/navbar/components/Navbar.test.jsx]: What are these empty strings for?

File: src/components/navbar/components/Navbar.jsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                  text
+                } = menu;
+                return (
+                  <Menu.Item key={key} className={className}>
+                    <Link className={linkClassName} to={linkTo}>
+                      {img ? <img src={img} alt={imgAlt} /> : text}
+                    </Link>
+                  </Menu.Item>
+                );
+              })}
```
Comment (Way2nnadi) [inline src/components/navbar/components/Navbar.jsx]: loving the map, good job. :)

Comment (edbella) [reply src/components/navbar/components/Navbar.jsx]: Thanks!

File: src/components/navbar/components/Navbar.test.jsx (hunk 6 above)
Comment (edbella) [reply src/components/navbar/components/Navbar.test.jsx]: It is my prettier code formatter that added it. I'll disable it.

File: src/App.js (hunk 5 above)
Comment (edbella) [reply src/App.js]: Ok. Will do that now.

File: .editorconfig [hunk 8]
```diff
@@ -9,7 +9,7 @@ root = true
 # Change these settings to your own preference
 [{*.js, *.jsx}]
 indent_style = tab
-indent_size = 4
+indent_size = 1
```
Comment (enyeInc) [inline .editorconfig]: why did you change this?

Comment (enyeInc) [reply .editorconfig]: Do not change any of the config files, without a good reason. indent size 4 is the convention that we are following here at Enye, to keep everyones code looking consistent

Comment (edbella) [reply .editorconfig]: Ok. I'll change it back.

Comment (enyeInc) [reply .editorconfig]: Sounds good, you will have a bunch of linting errors (mainly indents) to fix
